get_clang.sh

```diff
@@ -7,43 +7,19 @@
 # Clang is configured to be able to cross-compile to all the supported
 # architectures by default (see <clang path>/bin/llc --version) which is great,
 # but compiler-rt is included only for the host architecture. Therefore we need
-# to combine several packages into one, which is the purpose of this script.
+# a custom build of clang, which is achieved by [1].
+# This script retrieves a pre-compiled image from the Docker Hub and extracts
+# the compiler.
 
 [ "$1" ] || { echo "Usage: get_clang.sh version [path]"; exit 1; }
 
 VER=${1}
 DEST=${2:-./clang-${VER}}
-X86_64=clang+llvm-${VER}-x86_64-linux-gnu-ubuntu-16.04
-AARCH64=clang+llvm-${VER}-aarch64-linux-gnu
-ARMV7A=clang+llvm-${VER}-armv7a-linux-gnueabihf
 
-set -x
+ARCH=$(uname -m)
+set -e -x
 
-TMPDEST=${DEST}_tmp${RANDOM}
-
-if [ -e ${TMPDEST} ]; then
-  echo Error: ${TMPDEST} exists
-  exit 1
-fi
-
-function cleanup() {
-  rm -f ${X86_64}.tar.xz ${AARCH64}.tar.xz ${ARMV7A}.tar.xz
-  rm -rf ${AARCH64} ${ARMV7A}
-}
-
-trap "{ exit 2; }" INT
-trap cleanup EXIT
-
-(wget -nv https://github.com/llvm/llvm-project/releases/download/llvmorg-${VER}/${X86_64}.tar.xz && tar xf ${X86_64}.tar.xz) &
-pids=$!
-(wget -nv https://github.com/llvm/llvm-project/releases/download/llvmorg-${VER}/${AARCH64}.tar.xz && tar xf ${AARCH64}.tar.xz) &
-pids="$pids $!"
-(wget -nv https://github.com/llvm/llvm-project/releases/download/llvmorg-${VER}/${ARMV7A}.tar.xz && tar xf ${ARMV7A}.tar.xz) &
-pids="$pids $!"
-
-wait $pids || exit 1
-
-mv ${X86_64} ${TMPDEST} || exit 1
-cp ${AARCH64}/lib/clang/${VER}/lib/linux/* ${TMPDEST}/lib/clang/${VER}/lib/linux || exit 1
-cp ${ARMV7A}/lib/clang/${VER}/lib/linux/* ${TMPDEST}/lib/clang/${VER}/lib/linux || exit 1
-mv ${TMPDEST} ${DEST}
+id=$(docker create jforissier/clang-${VER}-${ARCH})
+docker cp $id:/root/clang-${VER} ${DEST}
+docker rm ${id}
+docker rmi jforissier/clang-${VER}-${ARCH}
```

Comment:
I suppose we need to Update the `Dockerfile` as well for [this](https://optee.readthedocs.io/en/latest/building/prerequisites.html#prerequisites), to show "docker", since our instructions actually is for people building from the host environment.

Reply:
Good catch. TBH I also find it a bit much to require Docker to simply download a toolchains package, especially since it brings some complications when running inside Docker such as in CI (add `-v /var/run/docker.sock:/var/run/docker.sock`, then if the user in the container is not root you need to map the docker group from the host).
Perhaps we could find a place to store a `.tar.bz2`?

Reply:
Never tried it, but I wonder if https://git-lfs.com/ is a solution for this problem?

Reply:
Ha! I'm just reading https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-git-large-file-storage :wink: but unfortunately there are very strict limits (1GB/month).
https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-storage-and-bandwidth-usage

Reply:
I believe I should first add the Clang build to this repository (such as add the proper Dockerfile and a new target `make clang-toolchains-build`). Then when we find a place to store the binary we can update `make clang-toolchains`. I will create a new PR for that.